packages/medusa/src/models/discount-rule-condition.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  DeleteDateColumn,
+  Entity,
+  Index,
+  JoinColumn,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from "typeorm"
+import { ulid } from "ulid"
+import {
+  CustomerGroup,
+  DiscountRule,
+  ProductCollection,
+  ProductTag,
+  ProductType,
+} from ".."
+import { DbAwareColumn, resolveDbType } from "../utils/db-aware-column"
+import { Product } from "./product"
+
+export enum DiscountRuleConditionType {
+  PRODUCTS = "products",
+  PRODUCT_TYPES = "product_types",
+  PRODUCT_COLLECTIONS = "product_collections",
+  PRODUCT_TAGS = "product_tags",
+  CUSTOMER_GROUPS = "customer_groups",
+}
+
+@Entity()
+export class DiscountRuleCondition {
+  @PrimaryColumn()
+  id: string
+
+  @DbAwareColumn({
+    type: "enum",
+    enum: DiscountRuleConditionType,
+  })
+  type: DiscountRuleConditionType
+
+  @Index()
+  @Column()
+  rule_id: string
```

Comment:
Should either be `discount_rule_id` or the join key name below should be changed to `rule_id`

Reply:
Good catch! Will change 👍